src/pages/Auth/CompleteRegistration/CompleteRegistration.tsx

```diff
@@ -1,70 +1,64 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import Wrapper, {
   RoleRadio,
   TitleText,
   FormBox,
   ApplyButton,
   ButtonText,
   RadioGroup,
-} from "pages/CompleteRegistration/CompleteRegistrationStyles";
+} from "pages/Auth/CompleteRegistration/CompleteRegistrationStyles";
 import useGoogleAuth from "hooks/useGoogleAuth";
 import useJwtDecoder from "hooks/useJwtDecoder";
 import { useTranslation } from "react-i18next";
-import { RadioChangeEvent } from "antd";
-import { useSelector, useDispatch } from "react-redux";
-import { RootState } from "store/store";
+import { useDispatch } from "react-redux";
 import { setUser } from "store/slices/userSlice";
 import { useNavigate } from "react-router-dom";
 import { WELCOME_ROUTE } from "utils/routeConsts";
 import { useUpdateUserMutation } from "services/user/setUserAPI";
 import { toast } from "react-toastify";
 
 export default function CompleteRegistration() {
-  const { role } = useSelector((state: RootState) => state.user);
-  const dispatch = useDispatch();
-  const { sub } = useJwtDecoder();
+  useGoogleAuth();
+  const [radioOption, setRadioOption] = useState();
+  const { sub: id } = useJwtDecoder();
 
+  const dispatch = useDispatch();
   const { t } = useTranslation();
   const navigate = useNavigate();
 
-  const [updateRole, { isSuccess, isError, isLoading, error }] =
-    useUpdateUserMutation();
-
-  useGoogleAuth();
+  const [updateRole, { isError, isLoading, error }] = useUpdateUserMutation();
 
   useEffect(() => {
     if (!isLoading && isError) {
       if ("status" in error!) {
         toast.error(t("CompleteRegistration.error"));
       }
-      return;
-    }
-    if (!isLoading && isSuccess) {
-      navigate(WELCOME_ROUTE);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isLoading]);
 
-  const setRole = (e: RadioChangeEvent) => {
+  const sendToDB = async () => {
+    navigate(WELCOME_ROUTE);
     dispatch(
       setUser({
-        role: e.target.value,
+        role: radioOption,
       })
     );
-  };
 
-  const sendToDB = () => {
-    updateRole({
-      id: sub,
-      role,
+    await updateRole({
+      id,
```

Comment:
Yes, but the backend has an endpoint that expects an ID in order to update the user. 
Since we can't store the ID and the token has the information we need, I'm using the hook useJwtDecoder() to decode and return the ID so I can pass it to the update role request.

Are you suggesting, for example, that I change the backend endpoints to instead of /settings-info/:id to /settings-info/self?
And then, since we're using passport, simply use the value we have stored in req.user.id to make queries to the db?